chromium-content-api/preload-chromium-content-api.c

```diff
@@ -0,0 +1,149 @@
+/* Copyright (C) 2020 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#define __USE_GNU
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <linux/limits.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+static int (*original_open)(const char *, int, ...);
+static int (*original_unlink)(const char *);
+
+
+void debug(char *s, ...)
+{
+	if (secure_getenv("SHM_CHROMIUM_DEBUG")) {
+		va_list va;
+		va_start(va, s);
+		fprintf(stderr, "SHM_CHROMIUM_WRAP: ");
+		vfprintf(stderr, s, va);
+		va_end(va);
+		fprintf(stderr, "\n");
+	}
+}
+
+const char *get_snap_name(void)
+{
+	const char *snapname = getenv("SNAP_INSTANCE_NAME");
+	if (!snapname) {
+		snapname = getenv("SNAP_NAME");
+	}
+	if (!snapname) {
+		debug("SNAP_NAME and SNAP_INSTANCE_NAME not set");
+	}
+	return snapname;
+}
+
+
+static char * redirect_shm(const char *snapname, const char *pathname)
+{
+	size_t max_len = PATH_MAX + 1;
+
+	// New path includes .snapname.
+	if ((strlen(snapname) + strlen(pathname) + 2) > max_len) {
```

Comment:
`>=` to account for null byte.

Reply:
If you're going to precompute the expected size, maybe just do that here and use it for the malloc/remainder?

Reply:
And account for "snap"